project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module

```diff
@@ -677,15 +683,171 @@ function c4m_user_access_denied() {
 /**
  * Implements hook_menu_alter().
  *
- * Make the user/%user/contact page just a callback.
+ * Make the user/%user/contact page.
  */
 function c4m_user_menu_alter(&$items) {
   $items['user/login'] = array(
     'title' => 'Please log in to continue',
   );
   $items['user/%user/contact'] = array(
     'type' => MENU_CALLBACK,
+    'title' => 'User contact form',
+    'description' => 'User contact form.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('c4m_user_contact_form', 1),
+    'access callback' => 'c4m_user_contact_form_access',
+    'access arguments' => array(1),
+  );
+}
+
+/**
+ * User contact form access function.
+ *
+ * Validates that user is logged in, and that account id entered at url exists.
+ */
+function c4m_user_contact_form_access($account) {
+  if (!user_is_logged_in()) {
+    // If not logged in, deny access.
+    return FALSE;
+  }
+
+  return !empty($account->uid);
+}
+
+/**
+ * Create 'Contact User' form.
+ */
+function c4m_user_contact_form($form, &$form_state, $account) {
+  global $user;
+  $form_state['addressee_user'] = $account;
+
+  $form['c4m_contact_user_from'] = array(
+    '#type' => 'textfield',
+    '#title' => t('From'),
+    '#description' => t('From address.'),
+    '#size' => 30,
+    '#default_value' => $user->mail,
+    '#disabled' => TRUE,
+  );
+
+  $form['c4m_contact_user_subject'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Subject'),
+    '#description' => t('Enter mail subject.'),
+    '#size' => 10,
+  );
+
+  $form['c4m_contact_user_body'] = array(
+    '#title' => t('Body'),
+    '#description' => t('Enter mail body.'),
+    '#type' => 'text_format',
+    '#format' => 'full_html',
```

Comment:
![screen shot 2016-07-11 at 16 40 58](https://cloud.githubusercontent.com/assets/789034/16734566/4ef00fd8-4786-11e6-9fc8-3a373539ae7c.png)
Only admins have access to full_html